src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7418,4 +7430,43 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_ParamsArrayInLambdaOnly_Title" xml:space="preserve">
     <value>Parameter has params modifier in lambda but not in target delegate type.</value>
   </data>
+  <data name="ERR_BadExtensionUnderlyingType" xml:space="preserve">
+    <value>Extension '{0}' cannot augment underlying type '{1}'. The extension underlying type may not be dynamic, a pointer, a nullable reference type, a ref struct or an extension.</value>
+  </data>
+  <data name="ERR_StaticBaseTypeOnInstanceExtension" xml:space="preserve">
+    <value>Instance extension '{0}' cannot augment underlying type '{1}' because it is static.</value>
+  </data>
+  <data name="ERR_OnlyBaseExtensionAllowed" xml:space="preserve">
+    <value>'{0}' is not an un-annotated extension type.</value>
+  </data>
+  <data name="ERR_PartialMultipleUnderlyingTypes" xml:space="preserve">
+    <value>Partial declarations of '{0}' must not specify different underlying types.</value>
```

Comment:
>underlying types [](http://example.com/codeflow?start=68&length=16)

Perhaps we shouldn't be adding a new concept. Could we instead say "extended type"? Or: "Partial declarations of '{0}' must not extend a different type."